src/server/node_services/nodes_monitor.js

```diff
@@ -205,6 +205,26 @@ class NodesMonitor extends EventEmitter {
         this.n2n_agent.set_rpc_address('n2n://nodes_monitor');
     }
 
+    waitForValue(accessor, pullInterval = 1000) {
+        let count = 1;
+        return new Promise(resolve => {
+            function pull() {
+                const val = accessor();
+                if (_.isUndefined(val)) {
+                    setTimeout(pull, pullInterval);
```

Comment:
Lets add a log message that we are still waiting for the system and how many retries?

Reply:
Regrading adding logs, We can.
Regarding how many retries, I don't see a value much for that. We can retry forever but we will stop at some point from desperation 🙂 

I started to doubt if this PR is actually needed... @dannyzaken WDYT?